### PR TITLE
Regression enhancement networkpolicy

### PR DESCRIPTION
**Pull request by qiliRedHat**
OCP-26279 - [BZ 1752636] Networkpolicy should be applied for large namespaces
https://polarion.engineering.redhat.com/polarion/redirect/project/OSE/workitem?id=OCP-26279

https://issues.redhat.com/browse/OCPQE-11379
Add kube-burner config to replace cluster-loader
Update script, remove cluster-loader, add assertion

### CODE REVIEW

**Comment by qiliRedHat** [comment]
Manual test bube-burner config, with 3 projects
```
# INDEXING=false WORKLOAD_TEMPLATE=/root/git/svt/perfscale_regerssion_ci/kubeburner-object-templates/pause-config.yml WORKLOAD=custom ./run.sh
Thu Jul 28 09:22:25 AM UTC 2022 creating python virtual environment at path: /tmp/5ab6f906-8845-432f-a3d9-9ae1480040e5/ripsaw-cli
Installing Operator from repo https://github.com/cloud-bulldozer/benchmark-operator.git and branch master
ripsaw-cli:ripsaw.clients.k8s:INFO :: benchmark-operator	benchmark-controller-manager-844dcdff69-sd7fg	Running
Operator Running
###############################################
Workload: custom
Workload template: /root/git/svt/perfscale_regerssion_ci/kubeburner-object-templates/pause-config.yml
Metrics profile: 
Alerts profile: 
QPS: 50
Burst: 50
UUID: 5ab6f906-8845-432f-a3d9-9ae1480040e5
Job iterations: 
###############################################
Thu Jul 28 09:22:55 AM UTC 2022 Creating kube-burner configmap
configmap/kube-burner-cfg-5ab6f906-8845-432f-a3d9-9ae1480040e5 created
Thu Jul 28 09:22:55 AM UTC 2022 Deploying benchmark
Starting Benchmark kube-burner-custom-5ab6f906-8845-432f-a3d9-9ae1480040e5, timeout set to 15000 seconds
ripsaw-cli:ripsaw.models.benchmark:INFO :: Benchmark kube-burner-custom-5ab6f906-8845-432f-a3d9-9ae1480040e5 created
ripsaw-cli:ripsaw.clients.k8s:INFO :: Waiting for state: Complete
ripsaw-cli:ripsaw.clients.k8s:INFO :: BENCHMARK	UUID					STATE
ripsaw-cli:ripsaw.clients.k8s:INFO :: kube-burner-custom-5ab6f906-8845-432f-a3d9-9ae1480040e5	Not Assigned Yet	
ripsaw-cli:ripsaw.clients.k8s:INFO :: kube-burner-custom-5ab6f906-8845-432f-a3d9-9ae1480040e5	5ab6f906-8845-432f-a3d9-9ae1480040e5	
ripsaw-cli:ripsaw.clients.k8s:INFO :: kube-burner-custom-5ab6f906-8845-432f-a3d9-9ae1480040e5	5ab6f906-8845-432f-a3d9-9ae1480040e5	Building
ripsaw-cli:ripsaw.clients.k8s:INFO :: kube-burner-custom-5ab6f906-8845-432f-a3d9-9ae1480040e5	5ab6f906-8845-432f-a3d9-9ae1480040e5	Running
ripsaw-cli:ripsaw.clients.k8s:INFO :: kube-burner-custom-5ab6f906-8845-432f-a3d9-9ae1480040e5	5ab6f906-8845-432f-a3d9-9ae1480040e5	Complete
ripsaw-cli:ripsaw.clients.k8s:INFO :: kube-burner-custom-5ab6f906-8845-432f-a3d9-9ae1480040e5 with uuid 5ab6f906-8845-432f-a3d9-9ae1480040e5 has reached the desired state Complete
Benchmark kube-burner-custom-5ab6f906-8845-432f-a3d9-9ae1480040e5 Complete
Run Metadata: 
{
    "name": "kube-burner-custom-5ab6f906-8845-432f-a3d9-9ae1480040e5",
    "namespace": "benchmark-operator",
    "uuid": "5ab6f906-8845-432f-a3d9-9ae1480040e5",
    "suuid": "5ab6f906",
    "status": "Complete"
}

Thu Jul 28 09:23:41 AM UTC 2022 Indexing benchmark metadata to https://search-perfscale-dev-chmf5l4sh66lvxbnadi4bznl3a.us-west-2.es.amazonaws.com:443/ripsaw-kube-burner
Thu Jul 28 09:23:41 AM UTC 2022 deactivating python virtual environment at path: /tmp/5ab6f906-8845-432f-a3d9-9ae1480040e5/ripsaw-cli
Thu Jul 28 09:23:41 AM UTC 2022 cleaning up unique run directory at path: /tmp/5ab6f906-8845-432f-a3d9-9ae1480040e5
(scale-ci) [root@ip-172-31-4-188 kube-burner]# oc get ns | grep large-network-policy
large-network-policy-1                             Active   3m32s
large-network-policy-2                             Active   3m32s
large-network-policy-3                             Active   3m32s
(scale-ci) [root@ip-172-31-4-188 kube-burner]# oc get all -n large-network-policy-1
NAME                             READY   STATUS    RESTARTS   AGE
pod/pause-1-1-7f7d5467b7-psvhn   1/1     Running   0          3m37s

NAME                TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)   AGE
service/pause-1-1   ClusterIP   172.30.200.100   <none>        80/TCP    3m37s

NAME                        READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/pause-1-1   1/1     1            1           3m37s

NAME                                   DESIRED   CURRENT   READY   AGE
replicaset.apps/pause-1-1-7f7d5467b7   1         1         1       3m37s
```

**Comment by qiliRedHat** [comment]
@paigerube14 PTAL

File: perfscale_regerssion_ci/kubeburner-object-templates/pause-config.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+jobs:
```
Comment (paigerube14) [inline perfscale_regerssion_ci/kubeburner-object-templates/pause-config.yml:1]: are these files being used in either of the scripts below?

File: perfscale_regerssion_ci/scripts/large_network_policy.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+## Auth=prubenda@redhat.com qili@redhat.com
+## Desription: Script for creating pause deployments and adding network policies
+## Polarion test case: OCP-26279 - [BZ 1752636] Networkpolicy should be applied for large namespaces
+## https://polarion.engineering.redhat.com/polarion/redirect/project/OSE/workitem?id=OCP-26279
+## Bug related: https://bugzilla.redhat.com/show_bug.cgi?id=1752636
+## Cluster config: 3 master (m5.2xlarge or equivalent) with 27 worker
+## kube-burner config: perfscale_regerssion_ci/kubeburner-object-templates/large-network-policy.yml
+################################################
+
+for i in $(oc get projects | grep large-network-policy | grep -Eo 'large-network-policy\S*');
```
Comment (paigerube14) [inline perfscale_regerssion_ci/scripts/large_network_policy.sh:12]: are you running kube-burner before this step? It is very confusing that the namespace creation isn't a part of this script

**Comment by paigerube14** [comment]
Talked with Qiujie offline and we decided it would be good to have a common script to clone the e2e-benchmark repo and do the ./run.sh (and needs to cd to the correct folder, can probably hard code to workload/kube-burner for now). Want to keep this as simple and robust as possible to be able to pass any file/object configuration in the future. 
With keeping it robust, we will want to set all the environment variables in the specific test run script, this includes the type of workload (will be custom in almost all cases). 

`
INDEXING=false WORKLOAD_TEMPLATE=my-config/kube-burner.cfg METRICS_PROFILE=my-metrics/metrics.yml ALERTS_PROFILE=my-alerts/alerts-profile.yml WORKLOAD=custom (etc) 
`

File: perfscale_regerssion_ci/scripts/large_network_policy.sh (hunk 2 above)
Comment (qiliRedHat) [reply perfscale_regerssion_ci/scripts/large_network_policy.sh:12]: @paigerube14 I want to discuss with you about how to call the custom kube-burner, from a script or making use of your existing kube-burner Jenkins job(may need some change). As we discussed yesterday, we both think script is better, which is clear and we can add all logic in a single test case entry script.

File: perfscale_regerssion_ci/kubeburner-object-templates/pause-config.yml (hunk 1 above)
Comment (qiliRedHat) [reply perfscale_regerssion_ci/kubeburner-object-templates/pause-config.yml:1]: Now this is used by the new added run_workload.sh to run kube-burner run.sh

**Comment by qiliRedHat** [comment]
@paigerube14 I updated based on what we talked yesterday. Please take a look again.

**Comment by nathan-weinberg** [comment]
@qiliRedHat couple small things:
1. `perfscale_regerssion_ci` needs to be fixed to `perfscale_regression_ci`
2. Can you add newlines to the end of these files?

File: perfscale_regerssion_ci/scripts/custom_workload_env.sh [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# Where to clone e2e-benchmarking from
+export E2E_BENCHMARKING_REPOSITORY=${E2E_BENCHMARKING_REPOSITORY:-"https://github.com/cloud-bulldozer/e2e-benchmarking"}
+export E2E_BENCHMARKING_BRANCH=${E2E_BENCHMARKING_BRANCH:-"master"}
+# Common ENVs of custom workload for kube-burner
+export WORKLOAD=${WORKLOAD:-"custom"}
+export INDEXING=${INDEXING:-"false"}
+export METRICS_PROFILE=${METRICS_PROFILE:-""}
+export ALERTS_PROFILE=${ALERTS_PROFILE:-""}
```
Comment (paigerube14) [inline perfscale_regerssion_ci/scripts/custom_workload_env.sh:8]: Should we add in the standard metrics profile under metrics-profiles? 

Might be also good to add in the defaults that are set up in our jenkins kube-burner job 
export COMPARISON_CONFIG=clusterVersion.json podLatency.json podCPU-avg.json podCPU-max.json podMemory-avg.json podMemory-max.json
export GSHEET_KEY_LOCATION=${GSHEET_KEY_LOCATION} (not sure how we can easily set this) 
export EMAIL_ID_FOR_RESULTS_SHEET=${EMAIL_ID_FOR_RESULTS_SHEET} (need to figure out how to easily set this) 
export GEN_CSV=true

File: perfscale_regerssion_ci/scripts/large_network_policy.sh [hunk 4]
```diff
@@ -0,0 +1,64 @@
+#/!/bin/bash
```
Comment (paigerube14) [inline perfscale_regerssion_ci/scripts/large_network_policy.sh:1]: Possible to add in folders under scripts to be able to separate out the type of functionality/test it is doing

Can set the below as placeholders? 
* network
* performance
* scalability

Comment (qiliRedHat) [reply perfscale_regerssion_ci/scripts/large_network_policy.sh:1]: Done

File: perfscale_regerssion_ci/scripts/custom_workload_env.sh (hunk 3 above)
Comment (qiliRedHat) [reply perfscale_regerssion_ci/scripts/custom_workload_env.sh:8]: Added 'metrics-profiles/metrics.yaml' as the default metrics profile.
For the rest, I think we can handle them in the Jenkins job like kube-burner Jenkins job does.

**Comment by qiliRedHat** [comment]
> @qiliRedHat couple small things:
> 
> 1. `perfscale_regerssion_ci` needs to be fixed to `perfscale_regression_ci`
> 2. Can you add newlines to the end of these files?

@nathan-weinberg Done

**Comment by qiliRedHat** [comment]
@paigerube14 I updated based on the above comments. Please take a look again.

**Pull request merged**